src/extensions/surveys.ts

```diff
@@ -628,19 +638,22 @@ export const createMultipleChoicePopup = (posthog: PostHog, survey: Survey, ques
             innerHTML: form,
         })
     }
-    formElement.addEventListener('change', () => {
-        const selectedChoices =
-            singleOrMultiSelect === 'single_choice'
-                ? formElement.querySelectorAll('input[type=radio]:checked')
-                : formElement.querySelectorAll('input[type=checkbox]:checked')
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore // TODO: Fix this, error because it doesn't recognize node list as an array
-        if ((selectedChoices.length ?? 0) > 0) {
-            ;(formElement.querySelector('.form-submit') as HTMLButtonElement).disabled = false
-        } else {
-            ;(formElement.querySelector('.form-submit') as HTMLButtonElement).disabled = true
-        }
-    })
+    if (!isOptional) {
+        ;(formElement.querySelector('.form-submit') as HTMLButtonElement).disabled = true
+        formElement.addEventListener('change', () => {
+            const selectedChoices =
```

Comment:
Does typing this as a `NodeList` not work? Wondering if we need the ts-ignore statements here

Reply:
You do it already for the results below